File: Stas/climbing-stairs.cs
// https://leetcode.com/problems/climbing-stairs/description/

public class Solution {
    public int ClimbStairs(int n) {

       if (n == 0)
         return 1;

       return n > 1 ? ClimbStairs(n - 1) + ClimbStairs(n -2) : n;
    }
}

File: Igor/algs/climbing-stairs.ts
// https://leetcode.com/problems/climbing-stairs/

function climbStairs(n: number): number {
  return fib(n);
}

function fib(n: number): number {
  let prev = 0;
  let current = 1;

  for (let i = current; i <= n; i++) {
    let next = prev + current;
    prev = current;
    current = next;
  }

  return current;
}
